File: java_compiler/test/Camp.py
class Node:
    def __init__(self, pointer = None):
        self.children = []
        if parent == None and pointer != None:
            self.pointer = pointer
            # self.parent = pointer.parent
            self.children = pointer.children
        elif pointer == None:
            self.pointer = self
            self.children = []


class Tree:
    def __init__(self):
        self.root = None
        self.cwd = self.root
    def insert(self, sub):
        if self.root == None:
            self.root = sub
            self.cwd = self.root
        else:
            sub.parent = self.cwd
            self.cwd.children.append(sub)



def main():
    t = Tree()
    s1 = Node("1")
    t.insert(s1)
    s2 = Node("2", s1)
    t.insert(s2)
    print(s1.children)

main()

File: java_compiler/src/test/Camp.java
package test;
import java.util.LinkedHashMap;

public class Camp {
    static class Node {
        Node pointer;
        Node parent;
        Node[] children;
        int value = 0;
        Node () {
            pointer = this;
            children = new Node[10];
        }
        Node (Node pointer) {
            this.pointer = pointer;
            children = pointer.children;
        }
    }
    static class Tree {
        Node root;
        Node cwd;
        void insert(Node sub) {
            if (root == null) {
                root = sub;
                cwd = root;
            } else {
                sub.parent = cwd;
                cwd.children[0] = sub;
            }
        }
    }
    public static void main(String[] args) {
        Tree t = new Tree();
		Node s1 = new Node();
		t.insert(s1);
		Node s2 = new Node(s1);
		t.insert(s2);
        System.out.println(s1.children);
        
        LinkedHashMap<String, Node> l = new LinkedHashMap<>();
        // Node s1 = new Node();
        l.put("key", s1);
        l.get("key").value = 2;
        System.out.println(s1.value);
    }
}